### DISCUSSION TITLE
Is it possible to disable downstream spans based off of a flag?

### DISCUSSION THREAD

**Discussion by simonzy15** [tokio-rs/tracing#2148] [General]
The problem I'm running into is finding an efficient way to deal with different configurations of incoming requests. Some requests need to have tracing enabled and some do not. The subscriber in this case is shared among these requests. Is there a simple way to disable and enable spans based off of this type of configuration?

**Comment by davidbarsky** [comment]
There are a few ways. I'll provide the different options:
- If you know at compilation time what requests/routes you'd like to have tracing for, I suggest creating a span for each request with the request path. I'd then use [`FilterFn`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.FilterFn.html#method.new) to match on the paths you're interested in. While a span will be created, it will not be _recorded_, which is what I think you're looking for.
- If you don't know how you want to filter ahead of time (e.g., there's a specific header you're filtering on that varies from request to request), I suggest looking at OpenTelemetry's [sampling rules](https://docs.rs/tracing-opentelemetry/latest/tracing_opentelemetry/trait.PreSampledTracer.html#tymethod.sampled_context).
- If that doesn't work, consider using [`DynFilterFn`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.DynFilterFn.html), but note that can have a _pretty bad_ performance impact in this use-case.

Can I ask how you're trying to filter spans from requests?

**Reply by simonzy15** [reply]
Thanks for the response!

As requests come in to our service, we have an enum that specifies their type. There's a specific type that should not be traced. So I'm trying to figure out a way to have the spans not record for a certain type.

**Reply by simonzy15** [reply]
@davidbarsky Would you happen to have any examples of using sampling rules to change the span to be enabled or disabled at runtime?
